packages/middleware-multipart/src/specs/multipart.middleware.spec.ts

```diff
@@ -0,0 +1,173 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as request from 'supertest';
+import { map } from 'rxjs/operators';
+import { r, use, httpListener, createContext } from '@marblejs/core';
+import { multipart$ } from '../multipart.middleware';
+import { streamFileTo } from '../multipart.util';
+
+const TMP_PATH = path.resolve(__dirname, '../../../../tmp');
+
+const memoryMultipart$ = r.pipe(
+  r.matchPath('/memory'),
+  r.matchType('POST'),
+  r.useEffect(req$ => req$.pipe(
+    use(multipart$()),
+    map(req => ({ body: {
+      file: req.file,
+      fields: req.body,
+    }}))
+  )));
+
+const memoryWithOptionsMultipart$ = r.pipe(
+  r.matchPath('/memory-with-options'),
+  r.matchType('POST'),
+  r.useEffect(req$ => req$.pipe(
+    use(multipart$({
+      maxFileCount: 1,
+      maxFieldCount: 2,
+      maxFileSize: 20,
+    })),
+    map(req => ({ body: {
+      file: req.file,
+      fields: req.body,
+    }}))
+  )));
+
+const filesystemMultipart$ = r.pipe(
+  r.matchPath('/filesystem'),
+  r.matchType('POST'),
+  r.useEffect(req$ => req$.pipe(
+    use(multipart$({
+      stream: streamFileTo(TMP_PATH),
+    })),
+    map(req => ({ body: {
+      file: req.file,
+      fields: req.body,
+    }}))
+  )));
+
+const app = httpListener({
+  effects: [memoryMultipart$, memoryWithOptionsMultipart$, filesystemMultipart$],
+}).run(createContext());
+
+afterEach(() => {
+  if (fs.existsSync(TMP_PATH)) { fs.rmdirSync(TMP_PATH); }
+});
+
+describe('multipart$', () => {
+  test('parses multipart/form-data request with additional fields and stores it in-memory', async () => {
+    const data = Buffer.from('test_buffer_1');
+
+    return request(app)
+      .post('/memory')
+      .field('field_1', 'test_1')
+      .field('field_2', 'test_2')
+      .attach('data_field', data)
+      .expect(200)
+      .then(({ body }) => {
+        // data_field
+        expect(body.file.data_field.size).toEqual(Buffer.byteLength(data));
+        expect(body.file.data_field.destination).toBeUndefined();
+        expect(body.file.data_field.filename).toBeUndefined();
+        expect(body.file.data_field.encoding).toEqual('7bit');
+        expect(body.file.data_field.mimetype).toEqual('application/octet-stream');
+        expect(body.file.data_field.fieldname).toEqual('data_field');
+        expect(data.equals(Buffer.from(body.file.data_field.buffer))).toBe(true);
```

Comment:
Why not `expect(data).toEqual(Buffer.from(body.file.data_field.buffer))`?

Reply:
You cannot assert buffers this way - the test will not pass.